lib/tessel/deploy.js

```diff
@@ -832,30 +833,72 @@ actions.project = function(options) {
 actions.compress = function(source) {
   source = typeof source === 'string' ? source : source.toString();
 
-  var content = uglify.parse(source, actions.compress.options);
+  var mangle = {
+    sort: true,
+    toplevel: true
+  };
+
+  var ast;
+
+  try {
+    // Attempt to use acorn, this will provide
+    // better support for modern Ecma-262
+    // (but unfortunately, not perfect)
+    ast = uglify.AST_Node.from_mozilla_ast(
+      acorn.parse(source, {
+        allowImportExportEverywhere: true,
+        allowReturnOutsideFunction: true,
+        ecmaVersion: 7,
+      })
+    );
+    // However, if it fails, also try uglify...
+  } catch (error) {
```

Comment:
Are there any kinds of errors returned from Acorn that might indicate that we should not continue?